proxy/director.go

```diff
@@ -126,7 +126,7 @@ func timedUnavailabilityFunc(wait time.Duration) func(*endpoint) {
 	return func(ep *endpoint) {
 		time.AfterFunc(wait, func() {
 			ep.Available = true
-			log.Printf("proxy: marked endpoint %s available", ep.URL.String())
+			log.Printf("proxy: marked endpoint %s available to retry it until refresh interval time arrived.", ep.URL.String())
```

Comment:
It will give up the endpoint after refresh interval. 
How about "marked endpoint %s available to give it another try".